src/codebert/codebert_model.rs

```diff
@@ -0,0 +1,360 @@
+}
+
+impl CodeBertMergesResources {
+    /// Shared under Apache 2.0 license by the Microsoft team at <https://github.com/microsoft/CodeBERT>. Modified with conversion to C-array format.
+    pub const CODEBERT: (&'static str, &'static str) = (
+        "codebert/merges",
+        "https://huggingface.co/microsoft/codebert-base/resolve/main/merges.txt",
+    );
+    /// Shared under Apache 2.0 license by the Hugging Face Inc. team at <https://huggingface.co/huggingface/CodeBERTa-language-id>. Modified with conversion to C-array format.
+    pub const CODEBERTA_LANG: (&'static str, &'static str) = (
+        "codeberta-language-id/merges",
+        "https://huggingface.co/huggingface/CodeBERTa-language-id/resolve/main/merges.txt",
+    );
+    /// Shared under Apache 2.0 license by the Microsoft team at <https://github.com/microsoft/CodeBERT>. Modified with conversion to C-array format.
+    pub const CODEBERT_MLM: (&'static str, &'static str) = (
+        "codebert-mlm/model",
+        "https://huggingface.co/microsoft/codebert-base-mlm/resolve/main/merges.txt",
+    );
+}
+
+pub struct CodeBertForFeatureExtraction {
+    codebert: BertModel<CodeBertEmbeddings>,
+}
+
+impl CodeBertForFeatureExtraction {
+    pub fn new<'p, P>(p: P, config: &BertConfig) -> CodeBertForFeatureExtraction
+    where
+        P: Borrow<nn::Path<'p>>,
+    {
+        let p = p.borrow();
+
+        let codebert = BertModel::<CodeBertEmbeddings>::new_with_optional_pooler(p, config, false);
+
+        CodeBertForFeatureExtraction { codebert }
+    }
+
+    pub fn forward_t(
+        &self,
+        input_ids: Option<&Tensor>,
+        mask: Option<&Tensor>,
+        token_type_ids: Option<&Tensor>,
+        position_ids: Option<&Tensor>,
+        input_embeds: Option<&Tensor>,
+        encoder_hidden_states: Option<&Tensor>,
+        encoder_mask: Option<&Tensor>,
+        train: bool,
+    ) -> CodeBertFeatureExtractionOutput {
+        let base_model_output = self
+            .codebert
+            .forward_t(
+                input_ids,
+                mask,
+                token_type_ids,
+                position_ids,
+                input_embeds,
+                encoder_hidden_states,
+                encoder_mask,
+                train,
+            )
+            .map(|transformer_output| {
+                (
+                    transformer_output.hidden_state,
+                    transformer_output.all_attentions,
+                )
+            });
+        let (tokens_embeddings, all_attentions) = base_model_output.unwrap();
+        CodeBertFeatureExtractionOutput {
+            hidden_states: tokens_embeddings,
+            all_attentions: all_attentions,
+        }
+    }
+}
+pub struct CodeBertFeatureExtractionOutput {
+    /// Hidden states for all intermediate layers
+    pub hidden_states: Tensor,
+    /// Attention weights for all intermediate layers
+    pub all_attentions: Option<Vec<Tensor>>,
+}
+pub struct CodeBertForFeatureExtractionModel {
```

Comment:
Looks like this code corresponds to a pipeline and not a model. We now have `SentenceEmbeddings` in this library, please refer to the relevant pipeline `sentence_embeddings`. Bert-like and RoBERTa-like models can be readily implemented from their masked-model implementation.

I believe you could just declare an alias, similar to https://github.com/guillaume-be/rust-bert/blob/59c0e668d68eb80c674100049ecd37c35c9ab4b5/src/roberta/roberta_model.rs#L997